### PR TITLE
[Frontend] Invoke Catalyst gradient functions from outside of `@qjit`

### PR DESCRIPTION
**Pull request by dime10**
The following is now supported:

```py
from catalyst import *

def f(x):
    return x**2

>>> grad(f)(4)
Array(8., dtype=float64, weak_type=True)
```

Support has been added to: `grad`, `jacobian`, `jvp`, `jvp`

[sc-46393]

### CODE REVIEW

File: doc/changelog.md [hunk 1]
```diff
@@ -42,6 +42,12 @@
 
 <h3>Improvements</h3>
 
+* Catalyst gradient functions ``grad``, ``jacobian``, ``jvp``, and ``vjp`` can now be invoked from
+  outside a ``@qjit`` context. This simplifies the process of writing functions where compilation
+  can be turned on and off easily by adding or removing the decorator. The functions dispatch to
+  their JAX equivalents when the compilation is turned off.
+  [(#375)](https://github.com/PennyLaneAI/catalyst/pull/375)
```
Comment (josh146) [inline doc/changelog.md:49]:
```suggestion
* Catalyst gradient functions `grad`, `jacobian`, `jvp`, and `vjp` can now be
  invoked from outside a `@qjit` context. This simplifies the process of
  writing functions where compilation can be turned on and off easily by adding
  or removing the decorator. The functions dispatch to
  their JAX equivalents when the compilation is turned off.
  [(#375)](https://github.com/PennyLaneAI/catalyst/pull/375)
```

**Comment by josh146** [comment]
@dime10 one documentation suggestion; the second sentence of the docstring of all 4 functions is

> This function allows the [gradient/Jacobian/JVP/VJP] of a hybrid quantum-classical function to be computed within the compiled program.

We should probably update it to

> This function allows the [gradient/Jacobian/JVP/VJP] of a hybrid quantum-classical function to be computed within the compiled program. Outside of a compiled function, this function will simply dispatch to its JAX counterpart [`jax.grad`/`jax.jacobian`/`jax.vjp`/`jax.jvp`].

File: doc/changelog.md (hunk 1 above)
Comment (dime10) [reply doc/changelog.md:49]: Quick side question, do you enforce a very narrow line width Josh (our standard is 100)?

Comment (dime10) [reply doc/changelog.md:49]: Haha true, what is the GitHub comment line length, 80?

Comment (dime10) [reply doc/changelog.md:49]:
```suggestion
* Catalyst gradient functions `grad`, `jacobian`, `jvp`, and `vjp` can now be invoked from
  outside a `@qjit` context. This simplifies the process of writing functions where compilation
  can be turned on and off easily by adding or removing the decorator. The functions dispatch to
  their JAX equivalents when the compilation is turned off.
  [(#375)](https://github.com/PennyLaneAI/catalyst/pull/375)
```
